### PR TITLE
Update comments for two operators.

### PR DESCRIPTION
**Pull request by lcy-seso**
- Codes of reorder_lod_tansor_by_rank_op have been updated but its comments has not been updated yet.
- Fix the display format of the shrink_rnn_memory_op and update the comments.

### CODE REVIEW

File: paddle/operators/reorder_lod_tensor_by_rank_op.cc [hunk 1]
```diff
@@ -26,22 +26,45 @@ class ReorderLoDTensorByRankTableOpProtoMaker
...
-The Out =  [Seq3, Seq0, Seq2, Seq1] with correct LoD information.
+If the indices stored in the Input(RankTable) is [3, 0, 2, 1], the
+Input(X) will be reordered that the forth sequence in Input(X) will become the
+first one, and then followed by the originally first, third, and the second one.
+
+This is:
+X = [Seq0, Seq1, Seq2, Seq3]. The indices in RankTable are [3, 0, 2, 1].
+Out =  [Seq3, Seq0, Seq2, Seq1] with a new LoD information.
+
+If the LoD information of Input(X) is empty, this means Input(X) is not a
```
Comment (guoshengCS) [inline paddle/operators/reorder_lod_tensor_by_rank_op.cc]: `not a sequence` may not be proper. Something like `not sequence data` might be better.

File: python/paddle/v2/fluid/layers/control_flow.py [hunk 2]
```diff
@@ -687,11 +687,10 @@ def topk(input, k):
 
 
 def lod_tensor_to_array(x, table):
-    """This function performs the operation that converts an LOD_Tensor to
-       an array.
+    """ Convert a LOD_TENSOR_ARRAY to an TensorArray.
```
Comment (guoshengCS) [inline python/paddle/v2/fluid/layers/control_flow.py]: Should it be `LOD_TENSOR_ARRAY` or `LoDTensor` here

File: python/paddle/v2/fluid/layers/control_flow.py [hunk 3]
```diff
@@ -721,11 +720,10 @@ def lod_tensor_to_array(x, table):
 
 
 def array_to_lod_tensor(x, table):
-    """This function performs the operations that converts an array to
-       an LOD_Tensor.
+    """Convert a LoD_Tensor_Aarry to an LoDTensor.
 
     Args:
-        x (Variable|list): The array that needs to be converted to a tensor.
+        x (Variable|list): The LoD Tensor Array to be converted to a tensor.
```
Comment (guoshengCS) [inline python/paddle/v2/fluid/layers/control_flow.py]: Should `converted to a tensor` be `converted to a LoDTensor` here

File: python/paddle/v2/fluid/layers/control_flow.py [hunk 4]
```diff
@@ -786,17 +785,24 @@ def increment(x, value=1.0, in_place=True):
...
+                           the position to which the input tensor will be
+                           written.
+        array (Variable|list): The output LOD_TENSOR_ARRAY to which the input
+                               tensor will be written. If this parameter is
+                               NONE, a new LOD_TENSOR_ARRAY will be created and
+                               returned.
+
     Returns:
-        Variable: The tensor type variable that has the data written to it.
+        Variable: The ouput LOD_TENSOR_ARRAY where the input tensor is written.
```
Comment (guoshengCS) [inline python/paddle/v2/fluid/layers/control_flow.py]: `The ouput` should be `The output`.

File: paddle/operators/reorder_lod_tensor_by_rank_op.cc [hunk 5]
```diff
@@ -26,22 +26,45 @@ class ReorderLoDTensorByRankTableOpProtoMaker
...
-NOTE: The RankTable does not need to be calculated by X.
+Input(X) is a batch of sequences. Input(RankTable) stores new orders of the
+input sequence batch. The reorder_lod_tensor_by_rank operator reorders the
+Input(X) according to the information provided by Input(RankTable).
 
 For example:
-The X = [Seq0, Seq1, Seq2, Seq3]. The indices of RankTable are [3, 0, 2, 1].
 
-The Out =  [Seq3, Seq0, Seq2, Seq1] with correct LoD information.
+If the indices stored in the Input(RankTable) is [3, 0, 2, 1], the
```
Comment (abhinavarora) [inline paddle/operators/reorder_lod_tensor_by_rank_op.cc]: is -> **are**

File: paddle/operators/reorder_lod_tensor_by_rank_op.cc [hunk 6]
```diff
@@ -26,22 +26,45 @@ class ReorderLoDTensorByRankTableOpProtoMaker
...
+Input(X) will be reordered that the forth sequence in Input(X) will become the
```
Comment (abhinavarora) [inline paddle/operators/reorder_lod_tensor_by_rank_op.cc]: forth -> **fourth**

File: paddle/operators/reorder_lod_tensor_by_rank_op.cc [hunk 7]
```diff
@@ -26,22 +26,45 @@ class ReorderLoDTensorByRankTableOpProtoMaker
...
+first one, and then followed by the originally first, third, and the second one.
```
Comment (abhinavarora) [inline paddle/operators/reorder_lod_tensor_by_rank_op.cc]: originally -> **original**

File: paddle/operators/reorder_lod_tensor_by_rank_op.cc [hunk 8]
```diff
@@ -26,22 +26,45 @@ class ReorderLoDTensorByRankTableOpProtoMaker
...
+
+This is:
+X = [Seq0, Seq1, Seq2, Seq3]. The indices in RankTable are [3, 0, 2, 1].
+Out =  [Seq3, Seq0, Seq2, Seq1] with a new LoD information.
+
+If the LoD information of Input(X) is empty, this means Input(X) is not a
+sequcence. This is also identical to a batch of sequences each sequence in which
```
Comment (abhinavarora) [inline paddle/operators/reorder_lod_tensor_by_rank_op.cc]: The sentence should be -> This is also identical to a batch of sequences where each sequence has a fixed length 1.

File: paddle/operators/reorder_lod_tensor_by_rank_op.cc [hunk 9]
```diff
@@ -26,22 +26,45 @@ class ReorderLoDTensorByRankTableOpProtoMaker
...
+has a fixed length 1. In this case, the reorder_lod_tensor_by_rank operator
+reorders each slice of Input(X) along the first axis according to
+Input(RankTable).
+
+This is:
+X = [Slice0, Slice1, Slice2, Slice3] and its LoD information is empty. The
+indices in RankTable are [3, 0, 2, 1].
+Out = [Slice3, Slice0, Slice2, Slice1] with no LoD information is appended.
+
+NOTE: This operator sorts Input(X) according to a given LoDRankTable which dose
```
Comment (abhinavarora) [inline paddle/operators/reorder_lod_tensor_by_rank_op.cc]: dose -> **does**

File: python/paddle/v2/fluid/layers/control_flow.py [hunk 10]
```diff
@@ -687,11 +687,10 @@ def topk(input, k):
 
 
 def lod_tensor_to_array(x, table):
-    """This function performs the operation that converts an LOD_Tensor to
-       an array.
+    """ Convert a LOD_TENSOR_ARRAY to an TensorArray.
 
     Args:
-        x (Variable|list): The tensor that needs to be converted to an array.
+        x (Variable|list): The lod tensor to be converted to a lod tensor array.
```
Comment (abhinavarora) [inline python/paddle/v2/fluid/layers/control_flow.py]: lod -> **LOD**

File: python/paddle/v2/fluid/layers/tensor.py [hunk 11]
```diff
@@ -146,25 +146,27 @@ def fill_constant(shape, dtype, value, out=None):
     """
     **fill_constant**
 
-    This function creates a tensor of specified *shape* and
-    *dtype*, and initializes this with a constant supplied in *value*.
+    This function creates a tensor with the specified `shape` and
```
Comment (abhinavarora) [inline python/paddle/v2/fluid/layers/tensor.py]: You can drop **the**

File: python/paddle/v2/fluid/layers/tensor.py [hunk 12]
```diff
@@ -146,25 +146,27 @@ def fill_constant(shape, dtype, value, out=None):
...
+    `dtype`, and initializes it with the constant specifed by `value`.
```
Comment (abhinavarora) [inline python/paddle/v2/fluid/layers/tensor.py]: the -> **a**

File: python/paddle/v2/fluid/layers/tensor.py (hunk 11 above)
Comment (lcy-seso) [reply python/paddle/v2/fluid/layers/tensor.py]: Done.

File: paddle/operators/reorder_lod_tensor_by_rank_op.cc (hunk 5 above)
Comment (lcy-seso) [reply paddle/operators/reorder_lod_tensor_by_rank_op.cc]: Done.

File: paddle/operators/reorder_lod_tensor_by_rank_op.cc (hunk 7 above)
Comment (lcy-seso) [reply paddle/operators/reorder_lod_tensor_by_rank_op.cc]: Done.

File: paddle/operators/reorder_lod_tensor_by_rank_op.cc (hunk 1 above)
Comment (lcy-seso) [reply paddle/operators/reorder_lod_tensor_by_rank_op.cc]: Done.

File: paddle/operators/reorder_lod_tensor_by_rank_op.cc (hunk 8 above)
Comment (lcy-seso) [reply paddle/operators/reorder_lod_tensor_by_rank_op.cc]: Done.

File: paddle/operators/reorder_lod_tensor_by_rank_op.cc (hunk 9 above)
Comment (lcy-seso) [reply paddle/operators/reorder_lod_tensor_by_rank_op.cc]: Done.

File: python/paddle/v2/fluid/layers/control_flow.py (hunk 2 above)
Comment (lcy-seso) [reply python/paddle/v2/fluid/layers/control_flow.py]: Done. thank you.

File: python/paddle/v2/fluid/layers/control_flow.py (hunk 10 above)
Comment (lcy-seso) [reply python/paddle/v2/fluid/layers/control_flow.py]: Done.

File: python/paddle/v2/fluid/layers/control_flow.py (hunk 4 above)
Comment (lcy-seso) [reply python/paddle/v2/fluid/layers/control_flow.py]: Done.

File: python/paddle/v2/fluid/layers/tensor.py (hunk 12 above)
Comment (lcy-seso) [reply python/paddle/v2/fluid/layers/tensor.py]: Done.

**Pull request merged**